### PR TITLE
Overhaul of example

### PR DESCRIPTION
**Pull request by WardLT**
- Stripped out unnecessary code
- Added a simple environment file
- Added documentation for how to run the code
- Made it work on GPU without removing IPU code
- Added a baseline model ran on GPU
- Modified GNN architecture to improve accuracy by ~3x

### CODE REVIEW

File: run_test.py [hunk 1]
```diff
@@ -72,78 +87,108 @@ def build_fn(atom_features: int = 64, message_steps: int = 8,
...
-                cb.ModelCheckpoint(os.path.join(test_dir, 'best_model.h5'), save_best_only=True),
-                cb.EarlyStopping(patience=128, restore_best_weights=True),
-                cb.CSVLogger(os.path.join(test_dir, 'train_log.csv')),
+                cb.ModelCheckpoint(test_dir / 'best_model.h5', save_best_only=True),
+                # We restart the best weights, but do not halt early to simplify timing across
+                cb.EarlyStopping(patience=args.num_epochs, restore_best_weights=True),
+                cb.CSVLogger(test_dir / 'train_log.csv'),
                 cb.TerminateOnNaN()
-            ], steps_per_epoch=10
+            ],
```
Comment (vksastry) [inline run_test.py:180]: steps_per_epoch is required for running on IPU. Otherwise all other changes looks ok. I am merging them to another branch and if it works fine on IPU- then we intend to merge the changes to the master.

Comment (WardLT) [reply run_test.py:180]: Sounds like a great plan. Sorry, I should have asked about the "steps_per_epoch" part as it seemed intentional?

Would you like me to detect the number of steps rather than keeping it fixed at a small number?

Comment (vksastry) [reply run_test.py:180]: yes, predetermining is the right way to do. Just to test that out initially, we use a fixed number.

Comment (WardLT) [reply run_test.py:180]: Makes sense :) I'll get you that change soon

**Comment by WardLT** [comment]
Alright. Fixed the steps per epoch. 🤞 everything works!

**Comment by vksastry** [comment]
Merged into develop branch
